x/authenticator/keeper/msg_server.go

```diff
@@ -19,21 +23,57 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 
 var _ types.MsgServer = msgServer{}
 
+// AddAuthenticator allows the addition of various types of authenticators to an account.
+// This method serves as a versatile function for adding diverse authenticator types
+// to an account, making it highly adaptable for different use cases.
+
 func (m msgServer) AddAuthenticator(
 	goCtx context.Context,
 	msg *types.MsgAddAuthenticator,
 ) (*types.MsgAddAuthenticatorResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// TODO: validate based on type of authenticator
-	// TODO: do we want to create and account for the data/pubkey
-	// here? Or in ante handler
-
 	sender, err := sdk.AccAddressFromBech32(msg.Sender)
 	if err != nil {
 		return nil, err
 	}
 
+	authenticators, err := m.Keeper.GetAuthenticatorsForAccount(ctx, sender)
+	if err != nil {
+		return nil, err
+	}
+
+	// If there are no other authenticators, ensure that the first authenticator is a SignatureVerificationAuthenticator.
+	if len(authenticators) == 0 && msg.Type != authenticator.SignatureVerificationAuthenticatorType {
+		return nil, fmt.Errorf("the first authenticator must be a SignatureVerificationAuthenticator")
```

Comment:
Note. When the FE is doing this, it could send a multimsg to add multiple authenticators. (sigVer for account, anything else)

Reply:
I think the messages will be processed one by one, so SigVer for account, then added to the authenticator store, then one for spend limits added to the authenticator store